src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs

```diff
@@ -59,6 +59,7 @@ public StringValues(string?[]? values)
         /// Defines an implicit conversion of a given string array to a <see cref="StringValues"/>.
         /// </summary>
         /// <param name="values">A string array to implicitly convert.</param>
+        [OverloadResolutionPriority(1)] // see https://github.com/dotnet/runtime/issues/101261
```

Comment:
@333fred, will this address the cited issue?

Reply:
For this to help, I think we'll need the first-class span work as well. Will need some careful testing. What I had suggested to Jeremy was to put attribute on one of the string.Join methods.

Reply:
Correct. I think ultimately, while you _could_ use this attribute to solve the problem on usage, the right solution will be first-class span. @jjonescz, we'll want to test this scenario explicitly (will clarify the issue high you offline).

Reply:
> the right solution will be first-class span

How does that solve it?

Reply:
> By allowing StringValues to be converted directly to ReadOnlySpan without needing 2 UDCs.

We decided we can't do that, unfortunately.

Reply:
> We decided we can't do that, unfortunately.

Huh?

Reply:
> > We decided we can't do that, unfortunately.
> 
> Huh? I'm saying first-class span will allow that.

And I'm saying we can't add an implicit cast operator for converting a StringValues to a ReadOnlySpan because we have no safe way to do it in a non-allocating way.

Reply:
> And I'm saying we can't add an implicit cast operator for converting a StringValues to a ReadOnlySpan because we have no safe way to do it in a non-allocating way.

No new operator is required. Because the array->ROS conversion will be a standard implicit conversion, StringValues will be convertible to ROS in an argument position through its existing string array UDC, even if we change nothing about the type.

Reply:
> Because the array->ROS conversion will be a standard implicit conversion

And that takes priority over the string->ROS conversion?

Reply:
Yes, because it is normal form. Just like the existing string[] overload, if the normal form is applicable, params won't be checked.

Reply:
Got it. In that case, great.